unidist/core/backends/mpi/core/worker/task_store.py

```diff
@@ -249,22 +250,31 @@ async def execute():
                             isinstance(output_data_ids, (list, tuple))
                             and len(output_data_ids) > 1
                         ):
+                            completed_data_ids = output_data_ids
                             for output_id, value in zip(output_data_ids, output_values):
                                 data_id = object_store.get_unique_data_id(output_id)
                                 object_store.put(data_id, value)
                         else:
                             data_id = object_store.get_unique_data_id(output_data_ids)
                             object_store.put(data_id, output_values)
+                            completed_data_ids = [data_id]
 
                         RequestStore.get_instance().check_pending_get_requests(
                             output_data_ids
                         )
+
+                # completed_data_ids = [common.unwrap_data_ids(arg) for arg in completed_data_ids]
+                data = {
+                    "operation_type": common.Operation.TASK_DONE,
+                    "output_data_ids": completed_data_ids,
+                }
                 # Monitor the task execution
                 # We use a blocking send here because we have to wait for
                 # completion of the communication, which is necessary for the pipeline to continue.
+
                 communication.mpi_send_object(
                     communication.MPIState.get_instance().comm,
-                    common.Operation.TASK_DONE,
+                    data,
                     communication.MPIRank.MONITOR,
```

Comment:
Let's use send_simple_operation here. On monitor side there should be two recvs: 1. recv_operation_type (without modifications), 2. recv_simple_operation in the if branch of task_done.